test/patterns/mutable_index/test_mutable_index.py

```diff
@@ -0,0 +1,74 @@
+import os
+from unittest import TestCase
+import unittest
+from pathlib import Path
+from aibolit.patterns.mutable_index.mutable_index import MutableIndex
+
+
+class TestMutableIndex(TestCase):
+    dir_path = Path(os.path.realpath(__file__)).parent
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_assignement(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexAssignment.java')),
+            'Should match simple index assignent'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_increment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexIncrement.java')),
+            'Should match index increment (++)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_decrement(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexDecrement.java')),
+            'Should match index decrement (--)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_addition_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexAdditionAssignment.java')),
+            'Should match index addition assignment (+=)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_subtraction_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexSubtractionAssignment.java')),
+            'Should match index subtraction assignment (-=)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_multiplication_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexMultiplicationAssignment.java')),
+            'Should match index multiplication assignment (*=)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_division_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexDivisionAssignment.java')),
+            'Should match index division assignment (/=)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_mod_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexModAssignment.java')),
```

Comment:
1) You have added tests for such operators as `i *= 2`, but you haven't added test for cases i  = i * 1 (excepting `i = i + 1`

So, pls add tests for the following cases:
`i = i - 1`
`i = i / 2`, etc. 

Consider all operators

2) Please add test with nested `for` cycles 
```
for (i = 1; i < 5; ++i ) {
 for (j = 1; j < 5; ++j) {
      i++;
  } 
} 
```

3) add test for post-fix operator
```
i--;
i++;
```

4) Please add test with nested `for` cycles and hiding the scope
```
for (i = 1; i < 5; ++i ) {
	for (i = 1; i < 5; ++i ) {
		i  = i + 1; // what cycle is changing a variable?
	} 
} 
```


4) Please add test with `for`, `while` cycle without block {} 
```
for (i = 1; i < 5; ++i ) 
   i = i - 1 
```


5) Assigning some value to variables is also changing the variable
```
for (i = 1; i < 5; ++i ) {
   int j = 1000;
}
```

Reply:
@lyriccoder Some considerations:

> 1. You have added tests for such operators as `i *= 2`, but you haven't added test for cases i  = i * 1 (excepting `i = i + 1`
> 
> So, pls add tests for the following cases:
> `i = i - 1`
> `i = i / 2`, etc.
> 
> Consider all operators
The expression is not important here, the important is the assignment; we don't need to test for all operations

> 
> 1. Please add test with nested `for` cycles
> 
> ```
> for (i = 1; i < 5; ++i ) {
>  for (j = 1; j < 5; ++j) {
>       i++;
>   } 
> } 
> ```
> 
> 1. add test for post-fix operator
> 
> ```
> i--;
> i++;
> ```
> 
> 1. Please add test with nested `for` cycles and hiding the scope
> 
> ```
> for (i = 1; i < 5; ++i ) {
> 	for (i = 1; i < 5; ++i ) {
> 		i  = i + 1; // what cycle is changing a variable?
> 	} 
> } 
> ```
> 
> 1. Please add test with `for`, `while` cycle without block {}
> 
> ```
> for (i = 1; i < 5; ++i ) 
>    i = i - 1 
> ```

Tests added

> 
> 1. Assigning some value to variables is also changing the variable
> 
> ```
> for (i = 1; i < 5; ++i ) {
>    int j = 1000;
> }
> ```
Assignment is already being tested in the first case.